### DISCUSSION TITLE
custom-icons-visual-studio-code-commands/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#622] [Comments]
# custom-icons-visual-studio-code-commands/

Learn the basics of using custom icons for VSCode commands and an important change in VSCode since version 1.83.0 how they are rendered.

https://www.eliostruyf.com/custom-icons-visual-studio-code-commands/

**Comment by boltex** [comment]
Thank you for this post about svg icons for commands in vscode as I'm trying to wrap my head around this lately :)

I'm also an extension author, and I'm having problems with enablement state of command icons that are displayed over treeview (view/title) and over editors (view/editor) I've posted this issue about the very problem I'm having : https://github.com/microsoft/vscode/issues/209397 

Reading your article, I wondered about the following:

What do you mean by "Since version 1.83.0, you no longer have to specify the light and dark properties. You can specify the relative path to the SVG file" ?
Do you mean you can omit the last folder part before the filename (the light/dark part)? Or that a single svg file suffice? 

Thanks! (oh and if you have any insights about the issue I raised on vscode about enablement that broke in the last one or two months, please go ahead! :) )

**Reply by estruyf** [reply]
Hi @boltex, I just looked into it, and it seems that in version 1.83.1, they reverted the change of how to load SVGs - https://github.com/microsoft/vscode/issues/194710

In version `1.83.0` they used a masking technique to color the icons. That way, you did not have to define a light and dark color version, but it gave issues where other image types or colors were used. That was the reason why they reverted the change.

About the issue you are facing, it has to do with this change: https://github.com/microsoft/vscode/commit/e0277c1feedcb5c6cfc8b897d2d9e969d6f7f9f4
